### DISCUSSION TITLE
Problem with ForEach parameter

### DISCUSSION THREAD

**Discussion by easiwriter** [pointfreeco/swift-composable-architecture#3484] [Q&A]
I've tried in vain to get this code to compile. For some reason it using a ForEach initialiser starting with init<V> and I don't understand why. If anyone can help I'd be very grateful. Here's the code:

First the View:

<img width="1174" alt="Screenshot 2024-11-11 at 13 55 09" src="https://github.com/user-attachments/assets/dcc634b2-18f6-4d99-a615-61526541e464">

Next the Reducer:

<img width="1174" alt="Screenshot 2024-11-11 at 13 55 53" src="https://github.com/user-attachments/assets/d26f1162-dd6a-4683-b481-6066ed7e2161">

**Comment by mbrandonw** [comment]
Hi @easiwriter, hard to say without code I can run myself, but the `$project` in the `ForEach` stands out. I think you should drop the $.

**Comment by easiwriter** [comment]
Hi, it still fails without the $. Here's the repo:

https://github.com/easiwriter/TCA-Write-

**Reply by easiwriter** [reply]
I've narrowed it down to the NavigationLink statement. Comment it out and it compiles, so it's not the ForEach. Why the compiler doesn't fail the NavigationLink instead is a mystery.
